packages/accordion/src/useAccordion.ts

```diff
@@ -5,185 +5,150 @@
-    }
-
-    setDisabledState(disabled);
-  };
-
-  const getHeaderProps = ({ role = 'heading', ariaLevel, ...props }: IHeaderProps = {}) => {
-    if (ariaLevel === undefined) {
-      throw new Error(
-        'Accessibility Error: You must apply the `ariaLevel` prop to the element that contains your heading.'
-      );
-    }
-
-    return {
-      role,
-      'aria-level': ariaLevel,
-      'data-garden-container-id': 'containers.accordion',
-      'data-garden-container-version': PACKAGE_VERSION,
-      ...props
-    };
-  };
-
-  const getTriggerProps = ({
-    index,
-    role = 'button',
-    tabIndex = 0,
-    ...props
-  }: ITriggerProps = {}) => {
-    if (index === undefined) {
-      throw new Error(
-        'Accessibility Error: You must provide an `index` option to `getTriggerProps()`'
-      );
-    }
-
-    sectionIndices.push(index);
-
-    return {
-      id: `${TRIGGER_ID}:${index}`,
-      role,
-      tabIndex,
-      'aria-controls': `${PANEL_ID}:${index}`,
-      'aria-disabled': disabledState.indexOf(index) !== -1,
-      'aria-expanded': isControlled
-        ? controlledExpandedState.includes(index)
-        : expandedState.includes(index),
-      onClick: composeEventHandlers(props.onClick, () => toggle(index)),
-      onKeyDown: composeEventHandlers(props.onKeyDown, (event: KeyboardEvent) => {
-        if (event.keyCode === KEY_CODES.SPACE || event.keyCode === KEY_CODES.ENTER) {
-          toggle(index);
-          event.preventDefault();
-        }
-      }),
-      ...props
-    };
-  };
-
-  const getPanelProps = ({ index, role = 'region', ...props }: IPanelProps = {}) => {
-    if (index === undefined) {
-      throw new Error(
-        'Accessibility Error: You must provide an `index` option to `getSectionProps()`'
-      );
-    }
-
-    return {
-      id: `${PANEL_ID}:${index}`,
-      role,
-      'aria-hidden': isControlled
-        ? !controlledExpandedState.includes(index)
-        : !expandedState.includes(index),
-      'aria-labelledby': `${TRIGGER_ID}:${index}`,
-      ...props
-    };
-  };
-
-  return {
-    getHeaderProps,
-    getTriggerProps,
-    getPanelProps,
-    expandedSections: controlledExpandedState,
-    disabledSections: disabledState
-  };
+
+      return {
+        role,
+        'aria-level': ariaLevel,
+        'data-garden-container-id': 'containers.accordion',
+        'data-garden-container-version': PACKAGE_VERSION,
+        ...props
+      };
+    },
+    []
+  );
+
+  const getTriggerProps = useCallback(
+    ({ index, role = 'button', tabIndex = 0, ...props }: ITriggerProps = {}) => {
+      if (index === undefined) {
+        throw new Error(
+          'Accessibility Error: You must provide an `index` option to `getTriggerProps()`'
+        );
+      }
+
+      return {
+        id: `${TRIGGER_ID}:${index}`,
+        role,
+        tabIndex,
+        'aria-controls': `${PANEL_ID}:${index}`,
+        'aria-disabled': disabledState.includes(index),
```

Comment:
If it's not disabled, I don't see any reason to render with this attribute. :) 

```suggestion
        'aria-disabled': disabledState.includes(index) || null,
```

Reply:
Updated!